jscomp/syntax/src/res_printer.ml

```diff
@@ -1591,9 +1591,13 @@ and print_label_declaration ~state (ld : Parsetree.label_declaration) cmt_tbl =
        ])
 
 and print_typ_expr ~(state : State.t) (typ_expr : Parsetree.core_type) cmt_tbl =
+  let parent_attrs =
+    let attrs = ParsetreeViewer.filter_parsing_attrs typ_expr.ptyp_attributes in
+    if Ast_uncurried.core_type_is_uncurried_fun typ_expr then attrs else []
+  in
   let print_arrow ?(arity = max_int) typ_expr =
     let attrs_before, args, return_type =
-      ParsetreeViewer.arrow_type ~arity typ_expr
+      ParsetreeViewer.arrow_type ~arity ~attrs:parent_attrs typ_expr
```

Comment:
Here's where the issue lies, the `@this` attribute is attached to the `Ptyp_constr` not `Ptyp_arrow`. Essentially what I'm doing here is just picking up the attributes from the `Ptyp_constr` and pass it as the initial attrs for the `ParsetreeViewer.arrow_type` to work with.